javascript_nodejs/50.contoso-cafe-bot/dialogs/shared/prompts/getUserNamePrompt.js

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+const { TextPrompt } = require('botbuilder-dialogs');
+const { MessageFactory } = require('botbuilder');
+const { LuisRecognizer } = require('botbuilder-ai');
+const { userProfileProperty } = require('../stateProperties');
+
+// LUIS service type entry for the get user profile LUIS model in the .bot file.
+const LUIS_CONFIGURATION = 'getUserProfile';
+
+// LUIS intent names from ./resources/getUserProfile.lu
+const WHY_DO_YOU_ASK_INTENT = 'Why_do_you_ask';
+const GET_USER_NAME_INTENT = 'Get_user_name';
+const NO_NAME_INTENT = 'No_Name';
+const NONE_INTENT = 'None';
+
+// User name entity from ./resources/getUserProfile.lu
+const USER_NAME = 'userName_patternAny';
+const TURN_COUNTER_PROPERTY = 'turnCounterProperty';
+
+// This is a custom TextPrompt that uses a LUIS model to handle turn.n conversations including interruptions.
+module.exports = class GetUserNamePrompt extends TextPrompt {
```

Comment:
This whole class feels inside out and bad pattern for prompts.  Isn't the prompt just supposed to be about validation? 

Why isn't this 
* GetUserNamePrompt extends Dialog
* it uses a TextPrompt internally
* The interruptions are dialogs which are invoked
* etc. ?